src/Logic/CoreRules.mli

```diff
@@ -49,6 +49,8 @@ exception NoOpenGoal
 
 val get_proof : proof_state -> proof_tree
 val mk_name : ?name:string -> sec_exp -> string
+val prove_by : rule -> tactic (*goal -> proof_state nondet*)
+                                           
```

Comment:
Exporting this function breaks the abstraction. Core tactics must be in CoreRules.